### PR TITLE
fix: fixed exist cropzone bug in toDataUrl image (fix #237 #224)

### LINKED ISSUE #237: Saving crop before hitting apply, saves the image with the crop guides visible
## Version
https://ui.toast.com/tui-image-editor

## Test Environment
Windows, Google Chrome 75.0.3770.142

## Current Behavior

https://www.screencast.com/t/iCwLRhSOfUf2

If you open up the image editor example on the website. Click on the crop tool, click one of the crop formats, and then click "download" without clicking "apply" it will result in downloading an image that still contains the crop guides. In this case it should either remove the guides prior to the download by cancelling the action or it should prevent the download. I tested some of the other tools, and they don't seem to suffer this same problem. In example rotate, flip didn't exhibit this behavior, but crop did.

Originally posted at https://github.com/nhn/tui.editor/issues/592 .

### PR DESCRIPTION
**Pull request by jinwoo-kim-nhn**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

#### issue
* https://github.com/nhn/tui.image-editor/issues/237
* https://github.com/nhn/tui.image-editor/issues/224

스크린샷 2020-01-15 오후 3 30 46

#### 해결
1. fabric.Rect 클래스를 상속받아서 Cropzone을 구현할 때 _render의 구현에서 ctx를 받아오는 방식이 fabricjs에서 가이드한 방법과 같지 않음 
    * (download시 깨진 cropzone영역이 생기는 버그의 원인 )
    * 수정후 toDataUrl 실행 후 Cropzone이 안깨지고 정상적으로 출력된 이미지를 얻음
2. toDataUrl 실행전에 cropzone을 숨겨주고 toDataUrl이 완료된후 cropzone을 다시 보여주도록 수정

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/component/cropper.js [hunk 1]
```diff
@@ -132,6 +132,21 @@ class Cropper extends Component {
         fabric.util.removeListener(document, 'keyup', this._listeners.keyup);
     }
 
+    /**
+     * Change cropzone visible
+     * @param {boolean} visible - cropzone visible state
+     */
+    changeVisible(visible) {
```
Comment (dongsik-yoo) [inline src/js/component/cropper.js]: I think this method name should be `changeVisibility`.

File: src/js/component/cropper.js [hunk 2]
```diff
@@ -132,6 +132,21 @@ class Cropper extends Component {
...
+        const cropzone = this._cropzone;
```
Comment (dongsik-yoo) [inline src/js/component/cropper.js]: There is this simple version implementation.
```js
this._cropzone && this._cropzone.set({visible});
```

Comment (jinwoo-kim-nhn) [reply src/js/component/cropper.js]: 넵 이렇게 하면 간결해지겠네요. ㅎㅎ 감사합니다

File: src/js/component/cropper.js (hunk 1 above)
Comment (jinwoo-kim-nhn) [reply src/js/component/cropper.js]: 네 이게 더 적확한 표현 같네요. 반영하겠습니다.

**Pull request merged**